opengrok-web/src/test/java/org/opengrok/web/api/v1/controller/HistoryControllerTest.java

```diff
@@ -0,0 +1,116 @@
+
+@ConditionalRun(RepositoryInstalled.GitInstalled.class)
+public class HistoryControllerTest extends JerseyTest {
+
+    private RuntimeEnvironment env = RuntimeEnvironment.getInstance();
+
+    private TestRepository repository;
+
+    @Rule
+    public ConditionalRunRule rule = new ConditionalRunRule();
+
+    @Override
+    protected Application configure() {
+        return new ResourceConfig(HistoryController.class);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        repository = new TestRepository();
+        repository.create(HistoryGuru.class.getResourceAsStream("repositories.zip"));
+
+        env.setSourceRoot(repository.getSourceRoot());
+        env.setDataRoot(repository.getDataRoot());
+        env.setProjectsEnabled(true);
+        env.setHistoryEnabled(true);
+        RepositoryFactory.initializeIgnoredNames(env);
+
+        Indexer.getInstance().prepareIndexer(
+                env,
+                true, // search for repositories
+                true, // scan and add projects
+                false, // don't create dictionary
+                null, // subFiles - needed when refreshing history partially
+                null); // repositories - needed when refreshing history partially
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+
+        // This should match Configuration constructor.
+        env.setProjects(new ConcurrentHashMap<>());
+        env.setRepositories(new ArrayList<>());
+        env.getProjectRepositoriesMap().clear();
+
+        repository.destroy();
+    }
+
+    @Test
+    public void testHistoryDTOEquals() {
+        HistoryEntry historyEntry = new HistoryEntry(
+                "1",
+                new Date(1245446973L / 60 * 60 * 1000),
+                "xyz",
+                null,
+                "foo",
+                true);
+        HistoryEntryDTO entry1 = new HistoryEntryDTO(historyEntry);
+        historyEntry.setAuthor("abc");
+        HistoryEntryDTO entry2 = new HistoryEntryDTO(historyEntry);
+
+        assertEquals(entry1, entry1);
+        assertNotEquals(entry1, entry2);
+
+        HistoryDTO history1 = new HistoryDTO(Collections.singletonList(entry1));
+        HistoryDTO history2 = new HistoryDTO(Collections.singletonList(entry2));
+        assertEquals(history1, history1);
+        assertNotEquals(history1, history2);
+    }
+
+    @Test
+    public void testHistoryGet() throws Exception {
+        final String path = "git";
+        Response response = target("history")
+                .queryParam("path", path)
+                .request()
+                .get();
+        HistoryDTO history = response.readEntity(new GenericType<HistoryDTO>() {});
+        History repoHistory = HistoryGuru.getInstance().getHistory(new File(repository.getSourceRoot(), path));
+        assertEquals(history, getHistoryDTO(repoHistory.getHistoryEntries()));
```

Comment:
can you assert something more - contanins some string, minimal length?
it relies on the serialization and when it's broken you wouldn't catch it.

Reply:
better now ? :-)